From a4f55f809faf1f600149a68e5952564793f94e83 Mon Sep 17 00:00:00 2001
From: Alan Tull <atull@kernel.org>
Date: Tue, 20 Feb 2018 09:56:58 -0600
Subject: [PATCH 030/151] FogBugz #536771: fpga: bridge: fix klocwork sprintf
 complaint

commit  e395a260945c93b0626b6af5a4750871958af8fc from
https://github.com/altera-opensource/linux-socfpga.git
branch is socfpga-5.4.64-lts

Klocwork give the following complaints:

/home/yves/linux/intel/linux-socfpga/drivers/fpga/fpga-bridge.c:243 --
-- SV.BANNED.REQUIRED.SPRINTF (4:Review) Analyze Function 'sprintf' is
deprecated.  Replace with more secure equivalent like 'sprintf_s', add
missing logic, or re-architect.

/home/yves/linux/intel/linux-socfpga/drivers/fpga/fpga-bridge.c:255 --
-- SV.BANNED.REQUIRED.SPRINTF (4:Review) Analyze Function 'sprintf' is
deprecated.  Replace with more secure equivalent like 'sprintf_s', add
missing logic, or re-architect.

Remedy these by using scnprintf rather than sprintf.

Signed-off-by: Alan Tull <atull@kernel.org>
Signed-off-by: Meng Li <Meng.Li@windriver.com>
Integrated-by: Jun Zhang <jun.zhang@windriver.com>
---
 drivers/fpga/fpga-bridge.c | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/drivers/fpga/fpga-bridge.c b/drivers/fpga/fpga-bridge.c
index 80bd8f1b2aa6..2f5214b4e116 100644
--- a/drivers/fpga/fpga-bridge.c
+++ b/drivers/fpga/fpga-bridge.c
@@ -292,7 +292,7 @@ static ssize_t name_show(struct device *dev,
 {
 	struct fpga_bridge *bridge = to_fpga_bridge(dev);
 
-	return sprintf(buf, "%s\n", bridge->name);
+	return scnprintf(buf, PAGE_SIZE, "%s\n", bridge->name);
 }
 
 static ssize_t state_show(struct device *dev,
@@ -304,7 +304,7 @@ static ssize_t state_show(struct device *dev,
 	if (bridge->br_ops && bridge->br_ops->enable_show)
 		enable = bridge->br_ops->enable_show(bridge);
 
-	return sprintf(buf, "%s\n", enable ? "enabled" : "disabled");
+	return scnprintf(buf, 10, "%s\n", enable ? "enabled" : "disabled");
 }
 
 static DEVICE_ATTR_RO(name);
-- 
2.26.1

